listenbrainz/labs_api/labs/api/recording_from_recording_mbid.py

```diff
@@ -54,45 +56,84 @@ def fetch(self, params, offset=-1, count=-1):
                     r = dict(row)
                     redirect_index[str(r['recording_mbid_old'])] = str(r['recording_mbid_new'])
 
+                # Now start looking up actual recordings
                 for i, mbid in enumerate(mbids):
                     if mbid in redirect_index:
                         mbids[i] = redirect_index[mbid]
 
                 query = '''SELECT r.gid AS recording_mbid, r.name AS recording_name, r.length, r.comment,
                                   ac.id AS artist_credit_id, ac.name AS artist_credit_name,
-                                  array_agg(a.gid) AS artist_credit_mbids
+                                  array_agg(a.gid) AS artist_credit_mbids,
+                                  first_release_date_year AS year
                              FROM recording r
                              JOIN artist_credit ac
                                ON r.artist_credit = ac.id
                              JOIN artist_credit_name acn
                                ON ac.id = acn.artist_credit
                              JOIN artist a
                                ON acn.artist = a.id
+                             JOIN track t
+                               ON t.recording = r.id
+                             JOIN medium m
+                               ON t.medium = m.id
+                             JOIN release rl
+                               ON m.release = rl.id
+                             JOIN release_group rg
+                               ON rl.release_group = rg.id
+                             JOIN release_group_meta rgm
+                               ON rg.id = rgm.id
                             WHERE r.gid
                                IN %s
-                         GROUP BY r.gid, r.id, r.name, r.length, r.comment, ac.id, ac.name
+                         GROUP BY r.gid, r.id, r.name, r.length, r.comment, ac.id, ac.name, rgm.first_release_date_year
                          ORDER BY r.gid'''
 
                 args = [tuple([psycopg2.extensions.adapt(p) for p in mbids])]
-                if count > 0:
-                    query += " LIMIT %s"
-                    args.append(count)
-                if offset >= 0:
-                    query += " OFFSET %s"
-                    args.append(offset)
-
                 curs.execute(query, tuple(args))
 
-                output = []
+                # Build an index of all the fetched recordings and clean up UUID() -> str
+                recording_index = {}
                 while True:
                     row = curs.fetchone()
                     if not row:
                         break
 
                     r = dict(row)
```

Comment:
I believe that if you use the DictCursor cursor_factory, you don't need to cast rows to a dict
Or is this because you need a real dict to be able to serialise the json?

Reply:
The latter.